server/core/region.go

```diff
@@ -1160,6 +1161,27 @@ func (r *RegionInfo) GetWriteLoads() []float64 {
 	}
 }
 
+// GetRangeCount returns the number of regions that overlap with the range [startKey, endKey).
+func (r *RegionsInfo) GetRangeCount(startKey, endKey []byte) int {
+	start := &regionItem{region: &RegionInfo{meta: &metapb.Region{StartKey: startKey}}}
+	end := &regionItem{region: &RegionInfo{meta: &metapb.Region{StartKey: endKey}}}
+	// it returns 0 if startKey is nil.
+	_, startIndex := r.tree.tree.GetWithIndex(start)
+	var endIndex int
+	var item rangetree.RangeItem
+	// it should return the length of the tree if endKey is nil.
+	if len(endKey) == 0 {
+		endIndex = r.tree.tree.Len() - 1
+	} else {
+		item, endIndex = r.tree.tree.GetWithIndex(end)
+		// it should return the endIndex - 1 if the endKey is the startKey of a region.
+		if item != nil && bytes.Equal(item.GetStartKey(), endKey) {
+			endIndex--
```

Comment:
Why not just return 1 here to make it more clear?

Reply:
It needs to consider the start key index if the endKey is equal the item startKey.

Reply:
Do we have a test case to test this kind of situation?

Reply:
yes， the first case(startkey="",endKey="") can be found in the stats_test.go.